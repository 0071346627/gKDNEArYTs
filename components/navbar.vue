<template>
  <nav class="navbar is-dark is-capitalized" role="navigation" aria-label="main navigation">
    <div class="container">
      <div class="navbar-brand">
        <h1 class="navbar-item is-size-5 has-text-weight-bold">
          {{ title }}
        </h1>
        <a
          :class="{'is-active': isActive}"
          @click="toggleMenu"
          role="button"
          class="navbar-burger burger"
          aria-label="menu"
          aria-expanded="false"
          data-target="navbar-menu"
        >
          <span aria-hidden="true" />
          <span aria-hidden="true" />
          <span aria-hidden="true" />
        </a>
      </div>
      <div id="navbar-menu" :class="{'is-active': isActive}" class="navbar-menu">
        <div class="navbar-start">
          <nuxt-link
            v-for="link in links"
            :key="link.name"
            :to="link.path"
            v-text="link.name"
            class="navbar-item"
          />
        </div>
      </div>
    </div>
  </nav>
</template>

<script>
export default {
  name: 'VNavbar',
  props: {
    title: {
      type: String,
      required: true
    },
    links: {
      type: Array,
      default: () => []
    }
  },
  data() {
    return {
      isActive: false
    }
  },
  methods: {
    toggleMenu() {
      this.isActive = !this.isActive
    }
  }
}
</script>
