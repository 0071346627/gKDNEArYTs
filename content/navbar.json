{
  "TITLE": "my weather",
  "LINKS": [{ "name": "home", "path": "/" }]
}
