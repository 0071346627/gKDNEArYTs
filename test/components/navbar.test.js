import Navbar from '~/components/navbar'
import { mount } from '@vue/test-utils'
import { LINKS as links, TITLE as title } from '~/content/navbar.json'

describe('Navbar', () => {
  let wrapper
  let html

  beforeAll(() => {
    wrapper = mount(Navbar, {
      propsData: { links, title },
      stubs: ['nuxt-link']
    })
    html = wrapper.html()
  })

  test('html snapshot', () => {
    expect(html).toMatchSnapshot()
  })

  test('navbar contains link tag for each provided link', () => {
    const tag = 'nuxt-link-stub'
    links.forEach(link => {
      expect(html).toContain(
        `<${tag} to="${link.path}" class="navbar-item">${link.name}</${tag}>`
      )
    })
  })

  test('clicking hamburger toggles isActive property', () => {
    wrapper.setData({ isActive: false })
    const hamburger = wrapper.find('.navbar-burger')
    expect(wrapper.vm.isActive).toBeFalsy()
    hamburger.trigger('click')
    expect(wrapper.vm.isActive).toBeTruthy()
    hamburger.trigger('click')
    expect(wrapper.vm.isActive).toBeFalsy()
  })

  test('toggling isActive property toggles the is-active class', () => {
    wrapper.setData({ isActive: false })
    const menu = wrapper.find('#navbar-menu')
    expect(menu.classes()).not.toContain('is-active')
    wrapper.setData({ isActive: true })
    expect(menu.classes()).toContain('is-active')
  })
})
