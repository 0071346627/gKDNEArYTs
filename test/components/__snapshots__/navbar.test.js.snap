// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar html snapshot 1`] = `
"<nav role=\\"navigation\\" aria-label=\\"main navigation\\" class=\\"navbar is-dark is-capitalized\\"><div class=\\"container\\"><div class=\\"navbar-brand\\"><h1 class=\\"navbar-item is-size-5 has-text-weight-bold\\">
        my weather
      </h1> <a role=\\"button\\" aria-label=\\"menu\\" aria-expanded=\\"false\\" data-target=\\"navbar-menu\\" class=\\"navbar-burger burger\\"><span aria-hidden=\\"true\\"></span> <span aria-hidden=\\"true\\"></span> <span aria-hidden=\\"true\\"></span></a></div> <div id=\\"navbar-menu\\" class=\\"navbar-menu\\"><div class=\\"navbar-start\\"><nuxt-link-stub to=\\"/\\" class=\\"navbar-item\\">home</nuxt-link-stub></div></div></div></nav>"
`;
